File: frontend/src/objects/merchant/MerchantApplicationStatus.ts
// Business note: frontend mirror of a backend service object; keep names and fields aligned unless this file is explicitly frontend-only.
export const APPLICATION_STATUS = {
  pending: 'Pending',
  approved: 'Approved',
  rejected: 'Rejected',
} as const

export type MerchantApplicationStatus =
  (typeof APPLICATION_STATUS)[keyof typeof APPLICATION_STATUS]

File: backend/src/main/scala/services/merchant/objects/MerchantApplicationStatus.scala
package services.merchant.objects

// Business note: service-owned domain object; mirror it in the frontend only when it is part of the protocol surface.
import system.objects.given
import system.objects.*

import io.circe.{Decoder, Encoder}

enum MerchantApplicationStatus:
  case Pending, Approved, Rejected

object MerchantApplicationStatus:
  private val enumLabel = text("MerchantApplicationStatus")
  def render(value: MerchantApplicationStatus): DisplayText = enumDisplayText(value)
  def parse(value: DisplayText): Option[MerchantApplicationStatus] = parseEnumValue(value, MerchantApplicationStatus.values)
  given Encoder[MerchantApplicationStatus] = enumEncoder
  given Decoder[MerchantApplicationStatus] = enumDecoder(MerchantApplicationStatus.values, enumLabel)
